arm_robots/src/arm_robots/cartesian.py

```diff
@@ -0,0 +1,221 @@
+import copy
+import enum
+
+import numpy as np
+import rospy
+from geometry_msgs.msg import PoseStamped
+from victor_hardware_interface_msgs.msg import ControlMode, MotionCommand
+
+
+def point_to_np_array(a):
+    return np.array([a.x, a.y, a.z])
+
+
+def pose_distance(a, b):
+    return np.linalg.norm(point_to_np_array(a.position) - point_to_np_array(b.position))
+
+
+class ArmSide(enum.IntEnum):
+    LEFT = 0
+    RIGHT = 1
+
+
+class CartesianImpedanceController:
+    def __init__(self, tf_buffer, motion_status_listeners, motion_command_publisher, joint_lim_low, joint_lim_high,
+                 position_close_enough=0.0025, timeout_per_1m=500, intermediate_acceptance_factor=7.,
```

Comment:
This is confusing, is this effectively a velocity limit? i.e minimum speed? The name is a bit confusing, I'd prefer `timeout_per_m` or `min_speed`

Reply:
also not sure such a timeout is a good idea. The rest of ROS contol uses the method of "trajectories have time info, so use one fixed timeout and add it to the expected time to determine timeout" That way it's effectively still based on distance, but you can just set it to 1s and not think about how that relates to the commanded speed of the motion.

Reply:
Not sure how to handle timeouts in a ROS consistent way if we don't have it inside the controller, since it's managed inside the `step` function.

Reply:
ok well we can revisit later if we decide we don't like this implementation